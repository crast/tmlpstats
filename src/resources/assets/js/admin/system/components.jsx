import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'

import Api from '../../api'
import {
    Form, formActions, SimpleField, SimpleFormGroup,
    SimpleSelect, NullableTextControl, NullableTextAreaControl, BooleanSelect } from '../../reusable/form_utils'
import { ALERT_TYPES } from '../../reusable/ui_basic'
import { objectAssign } from '../../reusable/ponyfill'
import debounceRender from '../../reusable/debounce'
import { connectRedux, rebind, delayDispatch } from '../../reusable/dispatch'
import { loadStateShape } from '../../reusable/shapes'
import { SystemMessages } from '../../reusable/system-messages'

import { systemMessagesData } from '../../lookups/lookups-system-messages'
const MODEL = 'admin.system.currentMessage'

// Avoid a potentially laggy set of constant re-rendering markdown
const DebouncedMessages = debounceRender(SystemMessages, 300, {maxWait: 4000})

@connectRedux()
export class AdminSystemMessages extends React.Component {
    static mapStateToProps(state) {
        const { currentMessage } = state.admin.system
        const { loadState, saveState } = state.lookups
        return {
            systemMessages: systemMessagesData.selector(state),
            loadState, saveState, currentMessage
        }
    }

    static propTypes = {
        systemMessages: PropTypes.instanceOf(Immutable.Map),
        loadState: loadStateShape,
        saveState: loadStateShape,
        currentMessage: PropTypes.object
    }

    constructor(props) {
        super(props)
        rebind(this, 'onSubmit', 'onSelect')
    }

    checkLoading() {
        const { systemMessages, loadState } = this.props
        if (systemMessages.size) {
            return true
        }
        if (loadState.available) {
            delayDispatch(this, systemMessagesData.manager.runNetworkAction('load', {}, {
                setLoaded: true,
                api: Api.Admin.System.allSystemMessages,
                successHandler(data, { dispatch }) {
                    let m = Immutable.Map()
                    data.forEach((d) => {
                        m = m.set(d.id, d)
                    })
                    dispatch(systemMessagesData.replaceItems(m))
                }
            }))
        }
        return false
    }

    render() {
        const { systemMessages, loadState, currentMessage } = this.props
        if (!this.checkLoading()) {
            return <div>Loading messages</div>
        }

        let items = []
        systemMessages.forEach((message) => {
            let query = 'All Regions'
            if (message.region) {
                query = `Region ${message.region}`
            } else if (message.center) {
                query = `Center ${message.center}`
            }
            items.push(
                <tr key={message.id}>
                    <td><a href="#" onClick={this.onSelect} data-mid={message.id}>{message.id}</a></td>
                    <td>{message.active? 'Y' : 'N'}</td>
                    <td>{message.section}</td>
                    <td>{query}</td>
                    <td>{message.title}</td>
                </tr>
            )
        })

        return (
            <div>
                <table className="table">
                    <thead>
                        <tr>
                            <td>ID</td>
                            <td>Active</td>
                            <td>Section</td>
                            <td>Shown For</td>
                            <td>Title</td>
                        </tr>
                    </thead>
                    <tbody>
                        {items}
                    </tbody>
                </table>
                <h4>{currentMessage.id? `edit message ${currentMessage.id}` : 'new message'}</h4>
                <Form model={MODEL} onSubmit={this.onSubmit} className="form-horizontal">
                    <SimpleField model={MODEL+'.section'} label="Section" />
                    <SimpleField model={MODEL+'.region'} label="Region ID" />
                    <SimpleFormGroup label="Center ID" divClass="col-md-3">
                        <NullableTextControl model={MODEL+'.center'} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Active" divClass="col-md-2">
                        <BooleanSelect model={MODEL+'.active'} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Level" divClass="col-md-3">
                        <SimpleSelect model={MODEL+'.level'} items={ALERT_TYPES} />
                    </SimpleFormGroup>
                    <SimpleField model={MODEL+'.title'} label="Title" />
                    <SimpleFormGroup label="Content (markdown)">
                        <NullableTextAreaControl model={MODEL+'.content'} rows={10} />
                    </SimpleFormGroup>
                    <SimpleFormGroup label="Preview">
                        <DebouncedMessages messages={Immutable.List.of(currentMessage)} />
                    </SimpleFormGroup>
                    <button type="submit" className="btn btn-primary">Save</button>
                </Form>
            </div>
        )
    }

    onSelect(e) {
        const id = e.target.dataset.mid
        let message = this.props.systemMessages.get(parseInt(id))
        if (message) {
            this.props.dispatch(formActions.load(MODEL, message))
        }
    }

    onSubmit(currentMessage) {
        // clone so we can fix
        currentMessage = objectAssign({}, currentMessage)
        const centerRegion = ['center', 'region']
        centerRegion.forEach((k) => {
            currentMessage[k] = currentMessage[k] || null  // Override with null
        })

        this.props.dispatch(systemMessagesData.manager.runNetworkAction('save', {data: currentMessage}, {
            api: Api.Admin.System.writeSystemMessage,
            successHandler(data, { dispatch }) {
                currentMessage.id = data.storedId
                dispatch(systemMessagesData.replaceItem(currentMessage))
                dispatch(formActions.load(MODEL, {active: true}))
            }
        }))
    }
}
