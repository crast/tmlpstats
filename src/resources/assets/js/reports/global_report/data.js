import Immutable from 'immutable'

import { TabbedReportManager } from '../tabbed_report/manager'
import Api from '../../api'

const _baseKey = Immutable.Record({
    regionAbbr: '',
    reportingDate: '',
    page: null
})

export class GlobalReportKey extends _baseKey {
    queryParams() {
        return {region: this.regionAbbr, reportingDate: this.reportingDate}
    }
}



export const reportData = new TabbedReportManager({
    prefix: 'reports/GlobalReport',
    findRoot: (state) => state.reports.global_report,
    actions: {
        load: {
            api: Api.GlobalReport.getReportPagesByDate
        }
    }
})
