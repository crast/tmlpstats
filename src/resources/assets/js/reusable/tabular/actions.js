export const CLICK_COLUMN = 'tabular/click-column'
export const CLICK_ITEM = 'tabular/click-item'
export const OVERRIDE_SORT = 'tabular/override-sort'


// clickColumn is used when we click a column, maybe with a modifier key.
export function clickColumn(table, column, modKey) {
    return {type: CLICK_COLUMN, table, column, modKey}
}

export function overrideSort(table, sorts) {
    return {type: OVERRIDE_SORT, table, sorts}
}
