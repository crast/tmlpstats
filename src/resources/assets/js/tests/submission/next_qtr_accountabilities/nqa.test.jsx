import React from 'react'
import renderer from 'react-test-renderer'

import { Provider, newTestingStore } from '../../testing-store'
import { setReportingDate } from '../../../submission/core/actions'
import { initializeApplications } from '../../../submission/applications/actions'
import { QuarterAccountabilitiesTable } from '../../../submission/next_qtr_accountabilities/components'
import { qtrAccountabilitiesData } from '../../../submission/next_qtr_accountabilities/data'

import { defaultSubmissionLookups } from '../core-lookups'

describe('NQA', () => {
    const store = newTestingStore()

    it('Should Render As Snapshot', () => {
        store.dispatch(setReportingDate('2017-04-07'))
        store.dispatch(initializeApplications({123: {id: 123, firstName: 'Person1', lastName: 'Person1Last'}}))
        store.dispatch(defaultSubmissionLookups())
        window.apiMocks['Submission.NextQtrAccountability.allForCenter'] = () => {
            return Promise.resolve([{id: 1}])
        }
        return store.dispatch(qtrAccountabilitiesData.load({center: 'DEN', reportingDate: '2017-04-07'})).then(() => {
            const params = {centerId: 'ABC', reportingDate: '2017-04-07'}
            const tree = renderer.create(
                <Provider store={store}>
                    <QuarterAccountabilitiesTable
                        params={params}
                        />
                </Provider>
            ).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
