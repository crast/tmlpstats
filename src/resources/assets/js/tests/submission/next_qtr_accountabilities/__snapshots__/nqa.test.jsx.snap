// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NQA Should Render As Snapshot 1`] = `
<form
  className="form-horizontal nextQuarterAccountabilities"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="panel panel-primary"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Team 1 Team Leader*
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.6.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.6.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.6.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-primary"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Team 2 Team Leader*
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.7.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.7.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.7.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-primary"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Statistician*
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.4.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.4.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.4.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-primary"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Logistics*
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.17.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.17.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.17.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Statistician Apprentice
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.5.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.5.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.5.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Access to Power
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.11.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.11.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.11.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Power to Create
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.12.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.12.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.12.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Game in the World
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.15.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.15.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.15.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        Landmark Forum
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.16.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.16.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.16.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        T1 Expansion
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.13.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.13.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.13.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        T2 Expansion
      </h3>
    </div>
    <div
      className="panel-body"
    >
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Choose From
        </label>
        <div
          className="col-md-8"
        >
          <div
            aria-label={undefined}
            className="btn-group"
            role="group"
          >
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Current Team Member
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Incoming T1/T2
            </button>
            <button
              className="btn btn-default"
              onClick={[Function]}
              type="button"
            >
              Other
            </button>
          </div>
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Person
        </label>
        <div
          className="col-md-8"
        >
          <div />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Email
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqEmail"
            name="submission.next_qtr_accountabilities.data.14.email"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Phone
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqPhone"
            name="submission.next_qtr_accountabilities.data.14.phone"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          Notes
        </label>
        <div
          className="col-md-8"
        >
          <input
            className="testing-control form-control nqNotes"
            name="submission.next_qtr_accountabilities.data.14.notes"
            type="text"
            value={undefined}
          />
        </div>
      </div>
      <div
        className=" form-group"
      >
        <label
          className="col-md-2 control-label"
        >
          
        </label>
        <div
          className="col-md-8"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
